### PR TITLE
chore: update react router to v6.4

### PR DESCRIPTION
**Pull request by flagrede**
## Description

Closes: regen-network/regen-registry#1312

- update react-router to `6.4.3`
- use `createBrowserRouter ` instead of `<BrowserRouter />` to leverage new Route Data API
- Refactor `RegistryNav` as `RegistryLayout`
- [App.tsx](https://github.com/regen-network/regen-web/pull/1540/files#diff-c510528c3ccdadd6305648a575d6e9a3b4dbb269310b82ebe39a95e135375158) now only contains routes declaration
- Sentry had to be temporarily disabled while waiting for the integration to be updated for RR v6.4 (related issue https://github.com/getsentry/sentry-javascript/issues/5872).

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] provided a link to the relevant issue or specification
- [x] provided instructions on how to test
- [x] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed
- [ ] once the PR is closed, set up backport PRs for `redwood`, `hambach` and optionally `master` (see below)

#### Setting up backport PRs

After merging your PR to `dev`, set up backports by doing the following:

1. If your branch does not have merge commits, add the following comment to
   your PR:

- If PR is a new feature: `@Mergifyio backport redwood hambach`.
- If PR is a bug fix: `@Mergifyio backport redwood hambach master`.

2. If your branch does have merge commits:

   a. Pull latest `dev`, `hambach` and `redwood` branches

   b. Create new branches for backports and merge `dev` (replace `<PR#>` with your PR #)

   ```
   git checkout -b hambach-backport-<PR#> hambach
   git merge dev
   git push origin hambach-backport-<PR#>
   git checkout -b redwood-backport-<PR#> redwood
   git merge dev
   git push origin redwood-backport-<PR#>`
   ```

   c. Open new PRs in regen-web targeting `hambach` and `redwood`, respectively.

   d. If PR is a bug fix repeat the same process for `master` branch.

### How to test

1.

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by flagrede** [comment]
@wgwz I reached out to the Sentry team and they started to work on Sentry support for React Router v6.4 today.

File: web-registry/src/pages/Admin/Admin.tsx [hunk 1]
```diff
@@ -1,38 +1,53 @@
...
+  const navigate = useNavigate();
+
+  const authError = error as OAuthError;
+  if (
+    authError &&
+    authError.error_description &&
+    authError.error_description.indexOf('email_not_verified:') > -1
+  ) {
+    const email: string = authError.error_description.split(':')[1];
+    navigate(`/verify-email?email=${email}`);
```
Comment (blushi) [inline web-registry/src/pages/Admin/Admin.tsx:20]: it's not being actively used on prod atm but this doesn't seem to work anymore (ie using `navigate` over `history.push`)

**Review by blushi — COMMENTED** [review]
Looks good overall, just a small comment

File: web-registry/src/pages/Admin/Admin.tsx (hunk 1 above)
Comment (flagrede) [reply web-registry/src/pages/Admin/Admin.tsx:20]: Thanks, will check.

**Review by victordelval — APPROVED** [review]
Looks good, thanks!

File: web-registry/src/pages/Admin/Admin.tsx (hunk 1 above)
Comment (flagrede) [reply web-registry/src/pages/Admin/Admin.tsx]: I revert that to the previous behavior.

**Review by blushi — APPROVED** [review]
@flagrede Please make sure to re-request my review once you've addressed my comments so I get notified, thanks ;)

**Comment by flagrede** [comment]
@Mergifyio backport redwood hambach

**Pull request merged**